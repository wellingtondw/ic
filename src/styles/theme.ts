export default {
  border: {
    radius: '0.625rem'
  },
  font: {
    main: {
      family: "'Montserrat', sans-serif",
      normal: 400,
      bold: 700
    },
    secondary: {
      family: "'Lato', sans-serif",
      bold: 700
    },
    sizes: {
      xsmall: '0.75rem',
      small: '0.875rem',
      normal: '1rem',
      medium: '1.125rem',
      large: '2.25rem',
      xlarge: '4.5rem'
    }
  },
  colors: {
    primary: '#2C97D1',
    black: '#2A2A2A',
    blackBg: '#2B2B2B',
    white: '#FFFFFF',
    yellow: '#FBFE63'
  },
  spacings: {
    xxxsmall: '0.25rem', // 4px
    xxsmall: '0.5rem', //8px
    xsmall: '1rem', //16px
    small: '1.5rem', //24px
    medium: '2rem', // 32px
    large: '2.5rem', //40px
    xlarge: '3rem', //48px
    xxlarge: '3.5rem' //56px
  },
  transition: {
    default: '0.3s ease-in-out',
    fast: '0.1s ease-in-out'
  }
} as const
