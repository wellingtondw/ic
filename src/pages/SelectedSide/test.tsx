import { screen } from '@testing-library/react'
import { renderWithTheme } from '../../utils/tests/helpers'
import theme from '../../styles/theme'

import { SelectedSide } from '.'
import { fireEvent } from '@testing-library/dom'

const mockGetYourSide = jest.fn()

jest.mock('../../assets/images/darth-vader.png', () => 'darth')
jest.mock('../../assets/images/luke-skywalker.png', () => 'luke')

jest.mock('../../components/Header', () => {
  return {
    Header: () => <div data-testid="Mock Header"></div>
  }
})

let mockPath = '/darth-vader'

jest.mock('react-router-dom', () => {
  return {
    useLocation: () => ({
      pathname: mockPath
    }),
    useNavigate: () => jest.fn()
  }
})

jest.mock('../../context/side', () => {
  return {
    useSide: () => ({
      getYourSide: mockGetYourSide,
      loading: false
    })
  }
})

describe('<SelectedSide />', () => {
  it('should be able to render correctly', () => {
    const { container } = renderWithTheme(<SelectedSide />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should be able to choose a side', () => {
    renderWithTheme(<SelectedSide />)

    const buttonEl = screen.getByRole('button', {
      name: /choose your path again, Padawan/i
    })

    fireEvent.click(buttonEl)

    expect(mockGetYourSide).toHaveBeenCalledTimes(1)
  })

  it('should be able to render with darth vader selected', () => {
    const { container } = renderWithTheme(<SelectedSide />)

    const buttonEl = screen.getByRole('button', {
      name: /choose your path again, Padawan/i
    })
    const imgEl = screen.getByRole('img')
    const masterEl = screen.getByText(/darth vader/i)

    expect(masterEl).toBeInTheDocument()

    expect(container.firstChild).toHaveStyle({
      background: theme.colors.blackBg,
      color: theme.colors.white
    })
    expect(buttonEl).toHaveStyle({
      background: theme.colors.white,
      color: theme.colors.black
    })
    expect(imgEl).toHaveAttribute('src', 'darth')
  })

  it('should be able to render with luke skywalker selected', () => {
    mockPath = 'luke-skywalker'
    const { container } = renderWithTheme(<SelectedSide />)

    const buttonEl = screen.getByRole('button', {
      name: /choose your path again, Padawan/i
    })
    const imgEl = screen.getByRole('img')
    const masterEl = screen.getByText(/luke skywalker/i)

    expect(masterEl).toBeInTheDocument()

    expect(container.firstChild).toHaveStyle({
      background: theme.colors.yellow,
      color: theme.colors.black
    })
    expect(buttonEl).toHaveStyle({
      background: theme.colors.black,
      color: theme.colors.yellow
    })
    expect(imgEl).toHaveAttribute('src', 'luke')
  })
})
