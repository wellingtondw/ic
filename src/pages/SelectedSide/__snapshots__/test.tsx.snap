// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectedSide /> should be able to render correctly 1`] = `
.c3 {
  width: 100%;
  max-width: 12.5rem;
  border-radius: 50%;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem;
  font-weight: bold;
  border: 0;
  border-radius: 0.625rem;
  position: relative;
  -webkit-transition: background-color 0.1s ease-in-out;
  transition: background-color 0.1s ease-in-out;
  background: #FFFFFF;
  color: #2A2A2A;
}

.c6:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c6:hover:not(:disabled) {
  background: #f7f7f7;
}

.c7 {
  visibility: visible;
}

.c0 {
  width: 100%;
  min-height: 100vh;
  background: #2B2B2B;
  color: #FFFFFF;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  height: calc(100vh - 4rem);
  padding: 0 2.5rem 1.5rem;
}

.c1 .c2 {
  margin-bottom: 2rem;
}

.c1 .c5 {
  margin-top: 2rem;
}

.c4 {
  font-size: 2.25rem;
  line-height: 44px;
  font-weight: 400;
}

.c4 strong {
  display: block;
}

@media (min-width:480px) {
  .c3 {
    max-width: 18.875rem;
  }
}

@media (min-width:768px) {
  .c3 {
    max-width: 23.75rem;
  }
}

@media (min-width:480px) {
  .c6 {
    padding: 1rem 2rem;
  }
}

@media (min-width:768px) {
  .c1 {
    height: auto;
  }

  .c1 .c2 {
    margin-bottom: 1rem;
    -webkit-order: 2;
    -ms-flex-order: 2;
    order: 2;
  }

  .c1 .c5 {
    -webkit-order: 1;
    -ms-flex-order: 1;
    order: 1;
    margin-top: 100px;
    margin-bottom: 4rem;
  }
}

@media (min-width:768px) {
  .c4 {
    -webkit-order: 3;
    -ms-flex-order: 3;
    order: 3;
  }
}

<div
  class="c0"
>
  <div
    data-testid="Mock Header"
  />
  <main
    class="c1"
  >
    <img
      class="c2 c3"
      src="darth"
    />
    <h1
      class="c4"
    >
      Your master is
      <strong>
        Darth Vader 
      </strong>
    </h1>
    <button
      class="c5 c6"
    >
      <div
        class="c7"
      >
        choose your path again, Padawan
      </div>
    </button>
  </main>
</div>
`;
